From bf4edc4ab848aecd8f282fb3dd9c7790785f22ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Thu, 13 Jun 2024 15:33:54 +0200
Subject: [PATCH] dt-bindings: perf: stm32: introduce DDRPERFM dt-bindings
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

DDRPERFM is the DDR Performance Monitor embedded in STM32MPU SOC.
It allows to monitor DDR events that come from the DDR Controller
such as read or write events.

Change-Id: Iade72928fa806720bea6628904684c6eda55e868
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/395141
Domain-Review: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
---
 .../bindings/perf/stm32-ddrperfm-pmu.yaml     | 103 ++++++++++++++++++
 1 file changed, 103 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/perf/stm32-ddrperfm-pmu.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/perf/stm32-ddrperfm-pmu.yaml
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/perf/stm32-ddr-pmu.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+maintainers:
+  - Clément Le Goffic <clement.legoffic@foss.st.com
+
+title: STMicroelectronics STM32 DDR Performance Monitor (DDRPERFM) bindings
+
+properties:
+  compatible:
+    enum:
+      - st,stm32-ddr-pmu
+      - st,stm32mp25-ddr-pmu
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    description: Reference clock for the DDR Performance Monitor
+    maxItems: 1
+
+  resets:
+    description: Reset control for the DDR Performance Monitor
+    maxItems: 1
+
+  access-controllers:
+    minItems: 1
+    maxItems: 2
+
+  st,dram-type:
+    description: type of DRAM on the associated memory controller
+    enum:
+      - 0 # LPDDR4
+      - 1 # LPDDR3
+      - 2 # DDR4
+      - 3 # DDR3
+
+required:
+  - compatible
+  - reg
+
+allOf:
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: st,stm32-ddr-pmu
+    then:
+      required:
+        - clocks
+        - resets
+
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: st,stm32mp25-ddr-pmu
+    then:
+      required:
+        - access-controllers
+        - st,dram-type
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/stm32mp1-clks.h>
+    #include <dt-bindings/reset/stm32mp1-resets.h>
+
+    ddrperfm: perf@5a007000 {
+        compatible = "st,stm32-ddr-pmu";
+        reg = <0x5a007000 0x400>;
+        clocks = <&rcc DDRPERFM>;
+        resets = <&rcc DDRPERFM_R>;
+        status = "disabled";
+    };
+
+  - |
+    ddrperfm: perf@48041000 {
+      compatible = "st,stm32mp25-ddr-pmu";
+      reg = <0x48041000 0x400>;
+      access-controllers = <&rcc 104>;
+      st,dram-type = <2>;
+      status = "disabled";
+    };
+
+  - |
+    #include <dt-bindings/clock/st,stm32mp21-rcc.h>
+    #include <dt-bindings/reset/st,stm32mp21-rcc.h>
+
+    ddrperfm: perf@480c0000 {
+      compatible = "st,stm32mp25-ddr-pmu";
+      reg = <0x480c0000 0x0 0x400>;
+      clocks = <&rcc CK_BUS_DDRPERFM>;
+      resets = <&rcc DDRPERFM_R>;
+      access-controllers = <&rifsc 67>;
+      st,dram-type = <0>;
+      status = "disabled";
+    };
+...
